src/single_random_draw.rs

```diff
@@ -0,0 +1,90 @@
+#[cfg(any(test, feature = "rand"))]
+use crate::Utxo;
+
+#[cfg(any(test, feature = "rand"))]
+use rand::{seq::SliceRandom, thread_rng};
+
+/// Randomly select coins for the given target by shuffling the utxo pool and
+/// taking UTXOs until the given target is reached, or returns None if the target
+/// cannot be reached with the given utxo pool.
+/// Requires compilation with the "rand" feature.
+#[cfg(any(test, feature = "rand"))]
+#[cfg_attr(docsrs, doc(cfg(feature = "rand")))]
+pub fn select_coins_srd<T: Utxo>(target: u64, utxo_pool: &mut [T]) -> Option<Vec<T>> {
```

Comment:
I figure it might also be a good time to talk about the `target`. You are aiming to make different attempts to come up with an input set. Some of these attempts will aim to create change and some will try to create a changeless transaction. Yet, the target should remain the same across all these attempts. That means that the target can only be composed of the parts that are the same for every coin selection attempt: the amount paid to recipients plus the fees for the recipient outputs plus the fees for the transaction header. After that we still add inputs, but due to us counting only their `effective_value`, they each already bring their own fees.—That leaves one thing unaccounted for: the change output. You may want to think about where the amount for the change and its fees should get added, and how you make sure it's present in algorithms that will create change while it is not present in algorithms that avoid change.

Reply:
@Xekyo sorry it's taken me a while to get back to you on this.  I believe this commit should address your comment: https://github.com/p2pderivatives/rust-bitcoin-coin-selection/pull/22/commits/542e2b236026f0b5591c5f0f99b59def7f34795c